### PR TITLE
qalculate-qt: New port

### PR DESCRIPTION
**Pull request by judaew**
###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 11.5 20G71 x86_64
Xcode 12.5.1 12E507

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [ ] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [ ] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by judaew** [comment]
During the build phase, I get the following error: [qalculate-qt-build.txt](https://github.com/macports/macports-ports/files/6900605/qalculate-qt-build.txt)

I don't quite understand what is causing the error. Can you help?

**Comment by kencu** [comment]
```
:info:build /Library/Developer/CommandLineTools/usr/bin/clang++ -c -pipe -stdlib=libc++ -I/opt/local/include -Os  -arch x86_64 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk -mmacosx-version-min=11.0 -Wall -Wextra -fPIC -DVERSION=\"3.20.1\" -DQT_NO_DEBUG -DQT_WIDGETS_LIB -DQT_GUI_LIB -DQT_NETWORK_LIB -DQT_CORE_LIB -I. -Isrc -I/opt/local/libexec/qt5/lib/QtWidgets.framework/Headers -I/opt/local/libexec/qt5/lib/QtGui.framework/Headers -I/opt/local/libexec/qt5/lib/QtNetwork.framework/Headers -I/opt/local/libexec/qt5/lib/QtCore.framework/Headers -Ibuild -I/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks/OpenGL.framework/Headers -I/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks/AGL.framework/Headers -I/opt/local/libexec/qt5/mkspecs/macx-clang -F/opt/local/libexec/qt5/lib -o build/itemproxymodel.o src/itemproxymodel.cpp
:info:build In file included from src/functionsdialog.cpp:12:
:info:build In file included from /opt/local/libexec/qt5/lib/QtWidgets.framework/Headers/QDialogButtonBox:1:
:info:build In file included from /opt/local/libexec/qt5/lib/QtWidgets.framework/Headers/qdialogbuttonbox.h:43:
:info:build In file included from /opt/local/libexec/qt5/lib/QtWidgets.framework/Headers/qtwidgetsglobal.h:43:
:info:build In file included from /opt/local/libexec/qt5/lib/QtGui.framework/Headers/qtguiglobal.h:43:
:info:build In file included from /opt/local/libexec/qt5/lib/QtCore.framework/Headers/qglobal.h:105:
:info:build /opt/local/libexec/qt5/lib/QtCore.framework/Headers/qcompilerdetection.h:590:6: error: Qt requires a C++11 compiler and yours does not seem to be that.
:info:build #    error Qt requires a C++11 compiler and yours does not seem to be that.
:info:build      ^
```

Different compilers have different default c++ standards if none is set on the build line. In this case, this version of clang does not default to c++11 or greater (it likely defaults to c++98).

Rather than rely on compiler defaults, the build should set the needed c++ version directly. We need to do that here to get this to build. If the needed c++ standard is unknown, you might try first ```-std=c++11``` and if that doesn't work ```-std=c++17```.

Getting qmake to listen to an added configure argument is a bit of a trick sometimes, however.

Sometimes this works:
```
configure.args-append CONFIG+=c++11
or
configure.args-append CONFIG+=c++17
```

Sometimes this is better:
```
configure.args-append  QMAKE_CXXFLAGS+=-std=c++11
or
configure.args-append  QMAKE_CXXFLAGS+=-std=c++17

File: math/qalculate-qt/Portfile [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    underneath. Features include customizable functions, units, arbitrary \
+    precision, plotting, and a user-friendly interface.
+
+checksums           rmd160 a6030704584f0fb3e709ab928bcf3ad869ac2799 \
+                    sha256 189e952b2b6b99ef63db46bd70d31f140adf19e8364d0eb4064daefca6a16e8a \
+                    size 852543
+
+depends_lib-append  \
+                    port:libqalculate \
+                    port:qt5-qtbase
```
Comment (kencu) [inline math/qalculate-qt/Portfile:28]: The qmake5 PortGroup automatically includes this dependency.

**Comment by kencu** [comment]
This worked:
```
compiler.cxx_standard 2011
configure.args-append QMAKE_CXXFLAGS+=-std=c++11
```

**Comment by kencu** [comment]
This worked too:
```
compiler.cxx_standard 2011
configure.args-append CONFIG+=c++11
```

**Comment by kencu** [comment]
The build leaves the app in a weird spot:
```
  /opt/local/bin/qalculate-qt.app/Contents/Info.plist
  /opt/local/bin/qalculate-qt.app/Contents/MacOS/qalculate-qt
  /opt/local/bin/qalculate-qt.app/Contents/PkgInfo
  /opt/local/bin/qalculate-qt.app/Contents/Resources/empty.lproj
  /opt/local/share/applications/qalculate-qt.desktop
  /opt/local/share/icons/hicolor/64x64/apps/qalculate-qt.png
```
You should either move that to ${applications_dir} or at least symlink it there.

**Comment by kencu** [comment]
It also looks like the Icon is not showing up, so that ideally would be fixed too:
Screen Shot 2021-07-30 at 8 26 35 PM

File: math/qalculate-qt/Portfile (hunk 1 above)
Comment (judaew) [reply math/qalculate-qt/Portfile]: Thanks

**Comment by judaew** [comment]
Thank you so much 🌟. So my thinking was correct, but I didn't know how to set `configure.args` correctly.

**Comment by judaew** [comment]
`qalculate-qt` requires `libqalculate` version 3.20 or higher. I have to wait until the PR https://github.com/macports/macports-ports/pull/11729 is merged and then I will update the HEAD of this branch.

**Comment by judaew** [comment]
Thanks so mush 🌟

**Comment by jbuonagurio** [comment]
> It also looks like the Icon is not showing up, so that ideally would be fixed too:
> Screen Shot 2021-07-30 at 8 26 35 PM

@kencu `qalculate-qt` now packages a MacOS-specific application icon. See PR: https://github.com/Qalculate/qalculate-qt/pull/98

The `post-build` and `destroot` sections in the portfile can be removed after the next release.

**Pull request merged**